subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmComponentDependencies.java

```diff
@@ -100,7 +100,7 @@ public <D extends ModuleDependency> D platform(D dependency) {
     @Override
     public <D extends ModuleDependency> D enforcedPlatform(D dependency) {
         if (dependency instanceof ExternalDependency) {
-            DeprecationLogger.whileDisabled(() -> ((ExternalDependency)dependency).setForce(true));
+            ((AbstractExternalModuleDependency)dependency).setForceForEnforcedPlatform(true);
```

Comment:
I think this needs to match what we're doing in DefaultDependencyHandler

Reply:
Done.